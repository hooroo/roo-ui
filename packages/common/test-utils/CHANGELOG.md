# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
