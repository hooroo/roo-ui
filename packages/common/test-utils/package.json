{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
