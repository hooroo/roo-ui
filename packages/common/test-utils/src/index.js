export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
