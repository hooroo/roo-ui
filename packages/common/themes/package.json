{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
