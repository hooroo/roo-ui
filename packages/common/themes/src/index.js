import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
