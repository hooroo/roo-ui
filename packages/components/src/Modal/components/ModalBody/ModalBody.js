import React from 'react';
import PropTypes from 'prop-types';

import { Box, Text } from '../../..';

const ModalBody = ({ children, ...props }) => (
  <Box {...props}>
    <Text fontSize="xl" color="text">
      {children}
    </Text>
  </Box>
);

ModalBody.displayName = 'Modal.body';

ModalBody.defaultProps = {
  textAlign: 'center', // eslint-disable-line react/default-props-match-prop-types
  px: 11, // eslint-disable-line react/default-props-match-prop-types
  pt: 7, // eslint-disable-line react/default-props-match-prop-types
  pb: 10, // eslint-disable-line react/default-props-match-prop-types
};

ModalBody.propTypes = {
  children: PropTypes.node.isRequired,
  ...Box.propTypes,
};

export default ModalBody;
