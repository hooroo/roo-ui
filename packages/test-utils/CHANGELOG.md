# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.50.7"></a>
## [0.50.7](https://github.com/hooroo/roo-ui/compare/v0.50.6...v0.50.7) (2018-08-14)


### Bug Fixes

* pass theme to child components; update snapshots ([#153](https://github.com/hooroo/roo-ui/issues/153)) ([1a83477](https://github.com/hooroo/roo-ui/commit/1a83477))




<a name="0.39.1"></a>
## [0.39.1](https://github.com/hooroo/roo-ui/compare/v0.39.0...v0.39.1) (2018-06-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.39.0"></a>
# [0.39.0](https://github.com/hooroo/roo-ui/compare/v0.38.4...v0.39.0) (2018-06-26)


### Features

* add build:watch scripts ([#98](https://github.com/hooroo/roo-ui/issues/98)) ([bcae0bf](https://github.com/hooroo/roo-ui/commit/bcae0bf))




<a name="0.32.0"></a>
# [0.32.0](https://github.com/hooroo/roo-ui/compare/v0.31.0...v0.32.0) (2018-06-15)


### Features

* add autocomplete component ([#77](https://github.com/hooroo/roo-ui/issues/77)) ([eee3f8c](https://github.com/hooroo/roo-ui/commit/eee3f8c))




<a name="0.19.0"></a>
# [0.19.0](https://github.com/hooroo/roo-ui/compare/v0.18.0...v0.19.0) (2018-05-22)


### Features

* Add <List/> & <ListItem> ([#49](https://github.com/hooroo/roo-ui/issues/49)) ([b922a8a](https://github.com/hooroo/roo-ui/commit/b922a8a))




<a name="0.7.0"></a>
# [0.7.0](https://github.com/hooroo/roo-ui/compare/v0.6.4...v0.7.0) (2018-05-11)


### Features

* add new icon set ([#14](https://github.com/hooroo/roo-ui/issues/14)) ([377c4c8](https://github.com/hooroo/roo-ui/commit/377c4c8))




<a name="0.5.4"></a>
## [0.5.4](https://github.com/hooroo/roo-ui/compare/v0.5.2...v0.5.4) (2018-05-08)


### Bug Fixes

* replace build script with prepublish ([90cafe3](https://github.com/hooroo/roo-ui/commit/90cafe3))




<a name="0.5.3"></a>
## [0.5.3](https://github.com/hooroo/roo-ui/compare/v0.5.2...v0.5.3) (2018-05-08)


### Bug Fixes

* replace build script with prepublish ([90cafe3](https://github.com/hooroo/roo-ui/commit/90cafe3))




<a name="0.5.0"></a>
# [0.5.0](https://github.com/hooroo/roo-ui/compare/v0.4.2...v0.5.0) (2018-05-07)


### Features

* add ui primitives ([#13](https://github.com/hooroo/roo-ui/issues/13)) ([5789b96](https://github.com/hooroo/roo-ui/commit/5789b96))




<a name="0.4.2"></a>
## [0.4.2](https://github.com/hooroo/roo-ui/compare/v0.4.0...v0.4.2) (2018-04-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.4.1"></a>
## [0.4.1](https://github.com/hooroo/roo-ui/compare/v0.4.0...v0.4.1) (2018-04-26)




**Note:** Version bump only for package @roo-ui/test-utils

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
