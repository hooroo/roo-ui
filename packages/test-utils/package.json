{
  "name": "@roo-ui/test-utils",
  "version": "0.5.3",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepublish": "babel src -d dist --copy-files"
  }
}
