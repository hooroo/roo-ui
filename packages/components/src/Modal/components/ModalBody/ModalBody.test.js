import React from 'react';
import { axe } from 'jest-axe';
import { mountWithTheme, shallowWithTheme } from '@roo-ui/test-utils';
import { qantas as theme } from '@roo-ui/themes';
import ModalBody from './ModalBody';

describe('<ModalBody />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mountWithTheme(
      <ModalBody>
        Lorem ipsum dolor sit amet
      </ModalBody>,
      theme,
    );
  });

  it('has no accessibility errors', async () => {
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });

  it('has display name', () => {
    expect(wrapper.find(ModalBody).name()).toBe('Modal.body');
  });

  it('renders correctly', () => {
    expect(shallowWithTheme(
      <ModalBody>
        Lorem ipsum dolor sit amet
      </ModalBody>,
      theme,
    )).toMatchSnapshot();
  });
});
