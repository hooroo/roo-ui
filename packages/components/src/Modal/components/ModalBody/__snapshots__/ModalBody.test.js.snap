// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalBody /> renders correctly 1`] = `
<Box
  blacklist={
    Array [
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "width",
      "fontSize",
      "color",
      "bg",
      "fontFamily",
      "textAlign",
      "lineHeight",
      "fontWeight",
      "fontStyle",
      "letterSpacing",
      "display",
      "maxWidth",
      "minWidth",
      "height",
      "maxHeight",
      "minHeight",
      "size",
      "ratio",
      "verticalAlign",
      "alignItems",
      "alignContent",
      "justifyItems",
      "justifyContent",
      "flexWrap",
      "flexBasis",
      "flexDirection",
      "flex",
      "justifySelf",
      "alignSelf",
      "order",
      "gridGap",
      "gridColumnGap",
      "gridRowGap",
      "gridColumn",
      "gridRow",
      "gridAutoFlow",
      "gridAutoColumns",
      "gridAutoRows",
      "gridTemplateColumns",
      "gridTemplateRows",
      "gridTemplateAreas",
      "gridArea",
      "border",
      "borderTop",
      "borderRight",
      "borderBottom",
      "borderLeft",
      "borderColor",
      "borderRadius",
      "boxShadow",
      "opacity",
      "overflow",
      "background",
      "backgroundImage",
      "backgroundPosition",
      "backgroundRepeat",
      "backgroundSize",
      "position",
      "zIndex",
      "top",
      "right",
      "bottom",
      "left",
      "textStyle",
      "colors",
      "variant",
      "theme",
    ]
  }
  pb={10}
  pt={7}
  px={11}
  textAlign="center"
>
  <Text
    blacklist={
      Array [
        "textStyle",
        "color",
        "bg",
        "fontSize",
        "fontWeight",
        "letterSpacing",
        "lineHeight",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "textAlign",
        "textDecoration",
        "display",
        "hidden",
      ]
    }
    color="text"
    fontSize="xl"
    hidden={false}
    textStyle="text"
  >
    Lorem ipsum dolor sit amet
  </Text>
</Box>
`;
